uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveRequestRightsTest.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_Service_LeaveManager as LeaveManagerService;
+use CRM_HRLeaveAndAbsences_Service_LeaveRequestRights as LeaveRequestRightsService;
+use CRM_HRCore_Test_Fabricator_Contact as ContactFabricator;
+
+/**
+ * Class CRM_HRLeaveAndAbsences_Service_LeaveRequestRightsTest
+ *
+ * @group headless
+ */
+class CRM_HRLeaveAndAbsences_Service_LeaveRequestRightsTest extends BaseHeadlessTest {
+
+  use CRM_HRLeaveAndAbsences_SessionHelpersTrait;
+  use CRM_HRLeaveAndAbsences_LeaveManagerHelpersTrait;
+
+  /**
+   * @var \CRM_HRLeaveAndAbsences_Service_LeaveRequestRights
+   */
+  private $leaveRequestRights;
+
+  /**
+   * @var array
+   *   Contact array
+   */
+  private $leaveContact;
+
+  /**
+   * @var array
+   *   Contact array
+   */
+  private $manager;
+
+  public function setUp() {
+    $leaveManagerService = new  LeaveManagerService();
+    $this->leaveRequestRights = new LeaveRequestRightsService($leaveManagerService);
+
+    $this->leaveContact = ContactFabricator::fabricate();
+    $this->manager = ContactFabricator::fabricate();
```

Comment:
Instead of creating database records, you can mock the `LeaveManagerService` and make it return the data you want. For example, when you want to test things for Admins, can you mock it and make the `currentUserIsAdmin()` method always return true. 
Here you can find a few more things about mocks https://phpunit.de/manual/5.7/en/test-doubles.html